packages/query-core/src/infiniteQueryBehavior.ts

```diff
@@ -114,6 +114,17 @@ export function infiniteQueryBehavior<TQueryFnData, TError, TData, TPageParam>(
 
         return result
       }
+      if (context.options.persister) {
+        context.fetchFn = () => {
+          return context.options.persister?.(fetchFn as any, {
+            queryKey: context.queryKey,
+            meta: context.options.meta,
+            signal: context.signal,
+          })
+        }
+      } else {
+        context.fetchFn = fetchFn
+      }
```

Comment:
this is a good solution because it will only wrap the whole `fetchFn` once, and not the fetch we do for every page 👍